modules/ROOT/pages/security/security.adoc

```diff
@@ -0,0 +1,68 @@
+= Security Aspects of ownCloud Infinite Scale
+:toc: right
+
+:description: This document gives you an overview of the different security measures of ownCloud Infinite Scale.
+
+== Introduction
+
+{description}
+
+== Operational aspects
+
+=== API Gateway
+
+oCIS makes use of a microservice architecture. But instead handling authorization and authentication in each service, we have implemented an API Gateway (the xref:deployment/services/proxy.adoc[proxy service]) which handles authentication and authorization. All requests have to go through this service. This has the nice effect that all other services can be hidden behind a firewall, which reduces the attack surface dramatically.
+
+=== Reverse Proxy
+
+Through the help of a reverse proxy in front of the API gateway, it's possible to implement further protection mechanisms like request rate limiting or blocking requests from sources which seem to be doing malicious actions like DOS or Slowloris attacks.
+
+=== Monitoring (Tracing, Log Aggregation)
+
+All oCIS services support request tracing. This means, you can visualize the path a request takes, which services are involved etc. In addition to that, all logs can be aggregated by common software like graylog, splunk, etc. This not only helps with administrating the services, but can also aid to detect malicious behavior by analyzing the logs or traces.
+
+=== OpenID Connect
+
+oCIS does not rely on a propriatary user management, instead it makes use of OpenID Connect. For small instances or testing purposes, a xref:deployment/services/idm.adoc[minimal LDAP Service named IDM] is included. For anything else, it is recommended that customers hook to their own OpenID Connect provider unsing the xref:deployment/services/idp.adoc[IDP service]. If an own OpenID Connect provider is not available, use secure standard software like Keycloak to connect to the existing user management system like Active Directory or any LDAP server.
+
+== Technological Aspects
+
+=== Role System
+
+oCIS provides a flexible role system which allows to customize the permissions of roles or to define custom roles in the system. This way, permissions for certain roles can be restricted to the absolute minimum.
+
+=== Extension System
+
+Compared to the apps/extensions in ownCloud 10, the oCIS extension/services system is more secure because they are developed and deployed as standalone services. They communicate via APIs with the core services and therefore can only use specific interfaces instead of having direct access to storages and data. A single malicious extension cannot corrupt the whole system.
```

Comment:
```suggestion
Compared to the apps/extensions in ownCloud 10, the Infinite Scale extensions/services are more secure because they are developed and deployed as standalone services. They communicate via APIs with the core services and therefore can only use specific interfaces instead of having direct access to storages and data. A single malicious extension cannot corrupt the whole system.
```